ert3/config/_stages_config.py

```diff
@@ -35,20 +52,40 @@ def is_executable(cls, location):
 
 class Step(_StagesConfig):
     name: str
-    script: List[str]
+    type: str
+    script: Optional[List[str]] = []
     input: List[InputRecord]
     output: List[OutputRecord]
-    transportable_commands: List[TransportableCommand]
+    transportable_commands: Optional[List[TransportableCommand]] = []
+    function: Optional[Callable]
+
+    def output_locations(self):
```

Comment:
I think these support functions are quite tailored towards the currently expected format of the `PrefectEvaluator`. My suggestion would be to move this data fetching into the `ert3.evaluator` module instead...